src/otaclient/ota_core/_updater_base.py

```diff
@@ -96,52 +141,34 @@ def __init__(
         )
         self._metrics.target_firmware_version = version
 
-        # ------ parse cookies ------ #
-        logger.debug("process cookies_json...")
-        try:
-            cookies = json.loads(cookies_json)
-            assert isinstance(
-                cookies, dict
-            ), f"invalid cookies, expecting json object: {cookies_json}"
-        except (JSONDecodeError, AssertionError) as e:
-            _err_msg = f"cookie is invalid: {cookies_json=}"
-            logger.error(_err_msg)
-            raise ota_errors.InvalidUpdateRequest(_err_msg, module=__name__) from e
-
-        # ------ parse upper proxy ------ #
-        logger.debug("configure proxy setting...")
-        self._upper_proxy = upper_otaproxy
-
         # ------ mount session wd as a tmpfs ------ #
         self._session_workdir = session_wd
         session_wd.mkdir(exist_ok=True, parents=True)
 
         # ------ init updater implementation ------ #
         self.ecu_status_flags = ecu_status_flags
 
-        # ------ init critical zone flags ------ #
-        self.critical_zone_flag = critical_zone_flag
-
         # ------ init variables needed for update ------ #
         _url_base = urlparse(raw_url_base)
         _path = f"{_url_base.path.rstrip('/')}/"
         self.url_base = _url_base._replace(path=_path).geturl()
 
         # ------ setup downloader ------ #
-        self._downloader_pool = _downloader_pool = DownloaderPool(
-            instance_num=cfg.DOWNLOAD_THREADS,
-            hash_func=sha256,
-            chunk_size=cfg.CHUNK_SIZE,
-            cookies=cookies,
-            # NOTE(20221013): check requests document for how to set proxy,
-            #                 we only support using http proxy here.
-            proxies={"http": upper_otaproxy} if upper_otaproxy else None,
-        )
-        self._download_helper = DownloadHelper(
-            downloader_pool=_downloader_pool,
-            max_concurrent=cfg.MAX_CONCURRENT_DOWNLOAD_TASKS,
-            download_inactive_timeout=cfg.DOWNLOAD_INACTIVE_TIMEOUT,
-        )
+        self._downloader_pool = downloader_pool
+
+
+class OTAUpdateOperatorInitLegacy(OTAUpdateOperatorInit):
+    ca_chains_store: CAChainStore
+
+
+class OTAUpdateOperatorLegacyBase(OTAUpdateOperator):
+    def __init__(
+        self,
+        *,
```

Comment:
Split a new `OTAUpdateOperatorLegacyBase` from the `OTAUpdateOperator` base class, and let this base do the OTA image spec version specific initializing(download helper setup, ca_chains_store).